src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs

```diff
@@ -181,7 +182,7 @@ public bool Equals(PackageSpec other)
                    EqualityUtility.DictionaryOfSequenceEquals(Scripts, other.Scripts) &&
                    EqualityUtility.DictionaryEquals(PackInclude, other.PackInclude, (s, o) => StringComparer.Ordinal.Equals(s, o)) &&
                    EqualityUtility.EqualsWithNullCheck(PackOptions, other.PackOptions) &&
-                   EqualityUtility.SequenceEqualWithNullCheck(TargetFrameworks, other.TargetFrameworks) &&
+                   TargetFrameworks.OrderedEquals(other.TargetFrameworks, fw => fw.FrameworkName.Framework, PathUtility.GetStringComparerBasedOnOS()) &&
```

Comment:
This won't work. 
If you do this you'd be comparing just the framework...so net45 and net46 would be considered equal. 

Additionally, it makes no sense to OS-based comparer on non-paths.

Reply:
yes, that should just check for complete framework and not just name, will change that.
ignorecase makes sense when project system itself change casing in nomination, it shouldn't be a valid case but just prevent against those unwanted failures.